File: python/algo11055.py
n = int(input())

seq = list(map(int, input().split()))

dp = [0 for _ in range(n)]
dp[0] = seq[0]
ans = dp[0]

for i in range(1, n):
    for j in range(i):
        if seq[j] < seq[i]:
            dp[i] = max(dp[i], dp[j])
    
    dp[i] += seq[i]
    ans = max(ans, dp[i])

print(ans)

File: c/algo11055.c
#include <stdio.h>

int max(int a, int b) {
    return a > b ? a : b;
}

int main() {
    int n, ans = 0;
    int dp[1001] = {0, };
    char input[1001];

    fgets(input, 1001, stdin);

    char *ptr = strtok(input, " ");
    int idx = 0;

    while (ptr != NULL) {
        for (int i = 0 ; i < idx ; i++) {
            if (input[i] < input[idx]) {
                dp[idx] = max(dp[idx], dp[i]);
            }
        }

        dp[idx] += input[idx];
        ans = max(ans, dp[idx]);

        ptr = strtok(NULL, " "); idx++;
    }

    printf("%d", ans);
    return 0;
}